### PR TITLE
Newstyle travis

### PR DESCRIPTION
**Pull request by dham**
This rewrites .travis.yml to be compatible with the new containerised travis. Allegedly this gets us access to more build resources.

### CODE REVIEW

File: docs/source/_static/bibliography.bib [hunk 1]
```diff
@@ -1,18 +1,30 @@
...
+  eprint={1505.03357},
+  url = {http://arxiv.org/abs/1505.03357}
+}
+
 @article{Rathgeber2015,
   title={Firedrake: automating the finite element method by composing abstractions},
   author={Rathgeber, Florian and Ham, David A and Mitchell, Lawrence and Lange, Michael and Luporini, Fabio and McRae, Andrew TT and Bercea, Gheorghe-Teodor and Markall, Graham R and Kelly, Paul HJ},
   archivePrefix = "arXiv",
-  journal={Submitted to ACM TOMS},
+  journal={Submitted to to ACM TOMS},
```
Comment (miklos1) [inline docs/source/_static/bibliography.bib]: Really?

File: .travis.yml [hunk 2]
```diff
@@ -8,6 +9,21 @@ notifications:
...
+        - git
+        - mercurial
+        - python-pip
+        - libopenmpi-dev
+        - openmpi-bin
+        - libblas-dev
+        - liblapack-dev
+        - gfortran
+        - libhdf5-mpi-dev
+        - swig
```
Comment (miklos1) [inline .travis.yml]: Can't we get rid of this after rename_ffc?

Comment (dham) [inline .travis.yml]: Yes, but right now we fail without it.

File: docs/source/_static/bibliography.bib [hunk 3]
```diff
@@ -22,9 +34,10 @@ @article{McRae2015
                   Mitchell, Lawrence and Ham, David A and Cotter,
                   Colin J},
   archivePrefix ="arXiv",
-  journal =      {Submitted to ACM TOMS},
+  journal =      {Submitted SIAM Journal on Scientific Computing},
```
Comment (miklos1) [inline docs/source/_static/bibliography.bib]: Missing "to"?

**Pull request merged**